Cherry pick  docs for 0.4.19, second part

Czaki opened the pull request:
# Description

Cherrypick a few new docs changes

melissawm committed:
Update docs contribution guide for two-repo setup (#5)

Co-authored-by: Lucy Liu <jliu176@gmail.com>
Co-authored-by: Juan Nunez-Iglesias <jni@fastmail.com>
Closes https://github.com/napari/docs/issues/3

chili-chiu committed:
add foundation grant onboarding workshop (#55)

melissawm committed:
Fixes formatting for the contributing documentation section (#79)

melissawm committed:
Move napari workshop template link to top of page (#90)

Closes https://github.com/napari/docs/issues/88

melissawm committed:
Improve makefile (#41)

# Description
Makes building paths more generic and allows building from a local
napari clone folder.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# References
Related to, but does not fix, #5222

jaimergp committed:
Add 'html-live' make action to support live reload workflows (#75)

# Description
This PR adds a `html-live` make action so we can write docs locally
without having to rebuild and serve manually. This depends on
`sphinx-autobuild`.

This is all you need:

```bash
$ make html-live
```

The incremental builds are still a bit slow, so it's not exactly "live"
but, it's indeed automated :) I've been using this in the background
while live previewing the markdown files on VS Code directly. Then, I
switch to the browser to double check.

## Type of change
- [X] Fixes or improves workflow, documentation build or deployment

melissawm committed:
Explain how to add new examples to the gallery (#137)

# Description
Adds some explanation on how to add new examples to the gallery.

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Fixes or improves existing content

# References
Closes #92

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
---------

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

melissawm committed:
Add instructions on how to use docs-xvfb (#138)

# Description
Adds a note on how to use docs-xvfb for faster/less annoying docs builds
:)

## Type of change
<!-- Please delete options that are not relevant. -->
- [x] Fixes or improves existing content

# References
<!-- What resources, documentation, and guides were used in the creation
of this PR? -->
<!-- If this is a fix or otherwise resolves an issue, reference it here
with "closes #(issue)" -->

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

---------

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

seankmartin committed:
Add instructions to build napari docs on Windows (#158)

# Description
Added instructions on how to build the napari docs on Windows using
either [Git Bash](https://gitforwindows.org/) or
[WSL(2)](https://learn.microsoft.com/en-us/windows/wsl/install).

## Type of change
- [x] Fixes or improves existing content

# References
closes #19

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

I think so? Another option is to make a cross platform makefile (see
[here](https://github.com/seankmartin/napari-docs/blob/cross_platform_make/Makefile))
but that seemed like it would introduce a lot of complexity to the
Makefile. Git bash and WSL both work very well for building the docs and
are probably good tools to get used to anyway.

---------

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>
Co-authored-by: Melissa Weber Mendonça <melissawm@gmail.com>

seankmartin committed:
Add more information to the documentation contribution guide (#157)

# Description

Adds some more information around the different napari documentation
resources and some more ways to build to the documentation contribution
guide.

Main things:
1. Breaks down the different sources of napari docs and what they are
written in (please let me know if I missed any!)
2. I've tried to highlight how you can contribute to the docs completely
through GitHub if you like with no local setup. I think this would be
great for simple changes and ease barrier to entry.
3. Mentioned the use of extra `make` commands such as `html-noplot` and
`linkcheck-files`.
4. I also modified the makefile slightly to accept filenames to link
check in the `linkcheck-files` command.

This is based on information I've picked up as I've been getting to
grips with how to efficiently contribute to docs.
It is possible some details are slightly inaccurate as a result, so
please let me know if anything I've written is a bit off!

I plan to make another PR at a later date with information about how to
build the docs from Windows using git bash or WSL, but I think it best
not to bloat this PR.

# References

Closes #34 - except for the comment "many docs changes will be fine to
make in the GH interface or a plaintext editor, but a really nice
feature of jupytext is that it allows you to run the .md files from a
Jupyter Lab or Jupyter Notebook interface. This is one of the main
reasons why we went with myst-md rather than plain notebooks. I think we
should explicitly encourage this way of writing docs, including a final
clear+run-all, as the preferred mode of contribution. Again, this could
be done in a follow-up."

I've not used jupytext before, so I wouldn't feel too comfortable adding
information about this. I'm not actually sure how running this document
(for example) in Jupyter helps with writing it so I'm very open to input
here! 😄

---------

Co-authored-by: Melissa Weber Mendonça <melissawm@gmail.com>
Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

Czaki committed:
Add information about constraint usage to install older napari release (#193)

# Description
Add information about the presence and usage of constraints files.

---------

Co-authored-by: Melissa Weber Mendonça <melissawm@gmail.com>
Co-authored-by: Juan Nunez-Iglesias <jni@fastmail.com>

tlambert03 committed:
make Talley emeritus SC (#204)

As discussed elsewhere, it's time for me to step down from this role.

Thanks @napari/core-devs

seankmartin committed:
Fix typo of points instead of shapes (#195)

# Description
A small typo fix, previously the "using the shapes layer" guide
mentioned the points layer.

## Type of change
- [x] Fixes or improves existing content

dstansby committed:
Improve titles of fundamentals tutorials (#196)

# Description
This makes the titles of the fundamental tutorials less verbose,
primarily by dropping "napari" - it's pretty obvious from context of
being in the napari docs that these pages are all about napari.

## Type of change
- [x] Fixes or improves existing content

jaimergp committed:
Update installation instructions for the new conda-forge packages and other changes in our packaging infra (#202)

# Description
Consolidated docs update for info about:

* https://github.com/conda-forge/napari-feedstock/pull/48
* https://github.com/napari/packaging/pull/75
* https://github.com/napari/packaging/pull/77

## Type of change
<!-- Please delete options that are not relevant. -->
- [X] Fixes or improves existing content

# References
* https://github.com/conda-forge/napari-feedstock/pull/48
* https://github.com/napari/packaging/pull/75
* https://github.com/napari/packaging/pull/77

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

---------

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

kne42 committed:
NAP 7: Key Binding Dispatch (#200)

NAP for how key bindings will be handled

---------

Co-authored-by: Juan Nunez-Iglesias <jni@fastmail.com>

aganders3 committed:
Use napari_scraper instead of qtgallery (#207)

# Description
This is a CI change that removes `qtgallery` in favor of a
napari-specific scraper. It's not that different but gives us a little
more control and is not much code.

This seems to fix the error seen in "Build PR Docs" for
https://github.com/napari/napari/pull/4865 and also speeds up the docs
build quite a bit (~11 min instead of ~25 min).

I'm no Qt expert but suspect the main improvements here are related to
adding `napari.Viewer.close_all()` (which maybe belongs in the reset fn)
and calling `processEvents()` one more time after this.

The main drawback right now is that this doesn't capture non-Viewer
windows, but this could probably be added if needed.

## Type of change
- [x] Fixes or improves workflow, documentation build or deployment

# References
closes #174 (maybe?)
fixes errors in docs build for
https://github.com/napari/napari/pull/4865

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas

melissawm committed:
Move contributing resources to top-level navbar (#208)

# Description
This PR moves the "Contributing" toctree to the top level navbar.

This has been discussed before, but we don't have an issue for it.

cc @jni @psobolewskiPhD

## Type of change
- [x] Fixes or improves existing content

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

melissawm committed:
Add roadmap board link to Roadmaps page (#212)

# Description

Adds link to GitHub board containing the napari roadmap to Roadmap page.

EDIT: [Rendered page is
here](https://output.circle-artifacts.com/output/job/31a35d46-3a9c-414e-ad63-7f7a8a988b93/artifacts/0/docs/_build/roadmaps/index.html)

## Type of change
- [x] Fixes or improves existing content

# References
Closes #182

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

seankmartin committed:
Fix getting started in napari linking to the unittest getting started page (#217)

Clicking the Getting started grid in the below image pointed to
https://docs.python.org/3/library/unittest.mock-examples.html#getting-started.

![image](https://github.com/napari/docs/assets/21295664/6a799ff0-1672-40d5-b263-4276d43beb88)

I've changed the link, so it *should* now link to
https://napari.org/dev/tutorials/fundamentals/getting_started.html,
which I assume was the intention here.

jni committed:
Update core developer list (#219)

- Add Kyle Harrington to the core developers: Kyle was invited about a
day ago
  and accepted!
- Move Ahmet Can to emeritus core developers: independently, Ahmet Can
reached
out to me asking to be moved. We thank Ahmet Can for his early
contributions
  to the project and wish him the best in his current and future work! 🙏

seankmartin committed:
Update Kyle's tag on the core devs page (#232)

# Description
I noticed that Kyle's tag on the core devs page was linking to the right
place, but the display name was wrong. So I've updated that here.

dstansby committed:
Update ndisplay title (#235)

# Description
Update the section title to reflect changes in the main repository.

Depends on https://github.com/napari/napari/pull/6192

## Type of change
- [x] Fixes or improves existing content

melissawm committed:
Fix titles on Getting Started section of user guide (#228)

# Description
Makes contents a bit clearer, and matches the tiles in the Getting
started index page.

h/t @seankmartin who called this out 😄

Old page:

![Screenshot_20230814_141217](https://github.com/napari/docs/assets/3949932/0d5b3014-a901-4a3a-8ff4-a3417248c83b)

New page:

![Screenshot_20230814_141249](https://github.com/napari/docs/assets/3949932/6bde0fb4-f2b4-4d42-bfdd-d6854eda712f)

## Type of change
- [x] Fixes or improves existing content

melissawm committed:
Add note on milestones for PRs (#221)

# Description
Adds a small note on adding milestones to PRs. If this is not useful,
feel free to close!

## Type of change
- [x] Fixes or improves existing content

dstansby committed:
Remove sub-sub section heading from rendering guide (#236)

# Description
The block of text under this heading doesn't refer to time-series data.
While it's applicable to time-series data, it's also applicable to 3D
volume data, so it seemed easiest just to remove the heading.

## Type of change
- [x] Fixes or improves existing content

lucyleeow committed:
DOC Minor fixes to nap 6 doc (#233)

# Description

- fix link to `npe2` code
- clarify that `Layer` would be a top level menu
- fix yaml example
- Add 'plugins' and 'layer context' menus  to 'contributable' menus list

melonora committed:
fix outdated dimension sliders documentation (#241)

# References and relevant issues
Closes #240

# Description
This PR updates the documentation under the section `Dimension Sliders`.
Here it was still shown that `current step` would return
a single integer and could be set by assigning a single integer. I have
checked whether this was also the case elsewhere in the documentation,
but that is not the case.

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>
Co-authored-by: Peter Sobolewski <pete.sd@gmail.com>

dgmccart committed:
Update napari-workshops.md (#243)

Adding links to 2 workshops hosted in September geared towards plugin
developers

# References and relevant issues
None

# Description
Adding 2 workshops geared towards plugin developers to the napari
workshops page

psobolewskiPhD committed:
Update selection instructions in Points tutorial for Shift-A keybinding (#238)

# References and relevant issues
Closes: https://github.com/napari/napari/issues/1760

# Description
This PR updates the Points tutorial to include the
select-all-points-in-layer (Shift-A) keybinding. Also, both keybindings
work in `(+)` mode, not just select mode.

Czaki committed:
Add nap for telemetry (#226)

# Description

Add NAP with the proposition of a telemetry system.

## Type of change
- [x] Adds new content page(s)

---------

Co-authored-by: Draga Doncila Pop <17995243+DragaDoncila@users.noreply.github.com>

psobolewskiPhD committed:
Update docs to suggest python 3.10 install (#246)

# References and relevant issues
In discussion about python version support, Matthias mentioned that more
napari downloads are 3.9 than 3.10
https://github.com/napari/napari/pull/5768#issuecomment-1742691809

# Description
This PR updates the `python_version` variable in conf.py to `3.10` that
is used for the myst substitution used in installation docs and
contributing (developer env) guide.

Czaki committed:
move to lfs

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
build docs

Czaki committed:
fix python version selection

Czaki committed:
fix constraints path

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

psobolewskiPhD committed:
Installation guide: Mention slow first launch time (#253)

# References and relevant issues
Closes https://github.com/napari/docs/issues/222

# Description
This PR adds a `Note` that first launch may be slow on some platforms
(macOS, Windows) and that anti-malware and security software may also
result in slower launches.
It's quite disconcerting when running `napari` doesn't visually do
anything immediately!

psobolewskiPhD committed:
Image layers can't have converted data type using contextual menu, only Labels (#252)

# References and relevant issues
Related to https://github.com/napari/napari/issues/6349
Note this doesn't close that, just makes it a true feature request.

# Description

Currently the viewer tutorial says the contextual menu can be used to
convert Image data type.
It cannot and not just because of the contextual menu setup, but because
the conversions are not implemented for image layers.
In this PR I fix this to say only Labels layers can be converted.

Czaki committed:
remove problematic trailig spaces

Czaki committed:
remove more spaces

Czaki committed:
Update docs/tutorials/fundamentals/viewer.md

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

psobolewskiPhD committed:
delete :orphan: (#269)

# References and relevant issues
First noted here:
https://github.com/napari/napari/pull/6384#discussion_r1385853123

# Description
There is an orphan `:orphan:` tag at the top of NAP-8. Pretty sure it's
not doing anything--the other naps don't have it. This deletes it.

haesleinhuepf committed:
add link to video from EMBO workshop (#273)

I'm proposing another link for the list of workshops:
Bioimage Analysis with Python and Napari (video lecture) as part of the EMBO Practical Course for Advanced Methods in Bioimage Analysis

---------

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

dstansby committed:
Improve flow of install page (#274)

# Description
Update installation guide:
- If I'm installing the bundled app, I don't care about pre-requisites
for installing the Python package, so move that section under the
"Installing as a Pytho package" section
- Modify the header levels, so the second level now only has "Install as
Python package", "Install as bundled app", "Next steps". In particular,
two sections which are only relevant to the Python package route have
been moved down a level to simplify the top level contents.

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
run ci

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

melissawm committed:
Update to use napari-sphinx-theme 0.3.0 (#267)

Prepares the docs repo to use the new napari-sphinx-theme, 0.3.0, which depends on pydata-sphinx-theme rather than forking it.

# References and relevant issues

Addresses napari/napari-sphinx-theme#113
Depends on napari/napari-sphinx-theme#134

tlambert03 committed:
Add Kyle to steering council, make Talley emeritus (#322)

@napari/core-devs, I've always felt that the core-devs list should
accurately reflect the _currently active_ team, and I don't see my
activity on napari returning any time soon, so this PR moves me from
"current" to "emeritus" core developers.

(It also adds a steering council asterisk for @kephale)

jni committed:
Update working groups leads (#327)

# References and relevant issues

Closes #324

See also [this Zulip thread](https://napari.zulipchat.com/#narrow/stream/298358-working-group-documentation/topic/update-docs-working-group-contacts/near/416509330)

# Description

Our working groups docs are woefully out of date. I've updated to
reflect the current makeup of the groups.

alisterburt committed:
move self to emeritus (#323)

I saw Talley's PR at #322 and thought I should probably do the same -
excited to see where the project goes, keep fighting the good fight and
I'll see you all around :-)

constantinpape committed:
Update annotate_points.md (#145)

# Description

This PR fixes issues with the point annotation tutorial:
- Make it work with latest napari version (tested with napari: 0.4.17,
magicgui: 0.7.2)
- Add `refresh_colors` to see changes when labels are updated

## Type of change
- [x] Fixes or improves existing content

# References
https://forum.image.sc/t/napari-point-annotation/80568/2

---------

Co-authored-by: Juan Nunez-Iglesias <jni@fastmail.com>
Co-authored-by: Andy Sweet <andrew.d.sweet@gmail.com>

Czaki committed:
docs requirements

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Czaki committed:
relax sphinx constraint

Czaki committed:
update docs dependencies

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

Czaki committed:
Move Nick and Loic to emeritus, sort emeritus core devs (#339)

# Description

Move Nicholas Sofroniew and Loic Royer to emeritus core developers and
sort this list alphabetically.

melissawm committed:
Close calendar event popover when clicking outside it (#337)

# References and relevant issues
Closes napari/napari-sphinx-theme#138, together with
https://github.com/napari/napari-sphinx-theme/pull/147

# Description
Clicking outside of the calendar event modal or pressing the ESC key
closes the modal.

---------

Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>

Czaki committed:
Rename `myModal` to `eventDetailBackground` in meetings schedule (#343)

# References and relevant issues

https://github.com/napari/docs/pull/337#discussion_r1472646332

# Description

Rename to increase code readability

jni committed:
Update conf.py to display the announcement banner (#342)

# References and relevant issues

Together with #341, this PR supersedes #332.

# Description

The announcement banner text html was added in #341. This PR now points
our
config to it, allowing the banner to be displayed.

Note: #332 also has a CSS fix. However, that fix is included in the
latest
release of the napari-sphinx-theme, so I think it's best to leave it out
of
this PR. (I would be happy to make a separate PR for it if we don't want
to
wait until the dependencies are updated.)

Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>

Czaki committed:
Merge branch 'v0.4.19x' into docs_0.4.19

melissawm reviewed (approved):
Docs artifacts look good, thanks!

kevinyamauchi committed:
add melissa to core devs (#345)

# References and relevant issues
This PR adds @melissawm , our newest core dev to the teams page!

# Description
<!-- What does this pull request (PR) do? Does it add new content,
improve/fix existing
context, improve/fix workflow/documentation build/deployment or
something else?
<!-- If relevant, please include a screenshot or a screen capture in
your content
change: "An image is worth a thousand words!" -->
<!-- You can use https://www.cockos.com/licecap/ or similar to create
animations. -->
<!-- You can also see a preview of the documentation changes you are
submitting by
clicking on "Details" to the right of the "Check the rendered docs
here!" check on your PR.-->

<!-- Final Checklist
- If images included: I have added [alt
text](https://webaim.org/techniques/alttext/)
If workflow, documentation build or deployment change:
- My PR is the minimum possible work for the desired functionality
- I have commented my code, to let others know what it does
-->

aganders3 committed:
Add NAP for multicanvas (#249)

Particular thanks to @brisvag and @melonora for previous work and
discussions on this, and for agreeing to be co-authors. I am happy to
absorb any criticism and direct praise their way 😄.

Thanks also to @andy-sweet and @Czaki, both of whom provided helpful
discussion while I worked on this NAP.

Of course I expect many changes to this and look forward to feedback.

---------

Co-authored-by: Andy Sweet <andrew.d.sweet@gmail.com>
Co-authored-by: Lorenzo Gaifas <brisvag@gmail.com>
Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>
